test/functional/feature_notifications.py

```diff
@@ -82,5 +83,36 @@ def run_test(self):
         self.log.info("-alertnotify should not continue notifying for more unknown version blocks")
         assert_equal(alert_text, alert_text2)
 
+        self.log.info("test -walletnotify with -walletnotifyconfirmations > 1")
+        n_confirmations = 6
+
+        # first have a different node generate blocks that we shouldn't see confirmations for
+        self.nodes[0].generate(n_confirmations + 1)
+        self.sync_all()
+        os.remove(self.tx_filename)
+
+        # now generate some blocks but don't bury any enough to be confirmed (they'll be confirmed later)
+        generated_before_restart = self.nodes[1].generate(n_confirmations - 2)
+        confirmed_txids = [t['txid'] for t in self.nodes[1].listtransactions("*", 100) if t['blockhash'] in generated_before_restart]
+
+        # restart node with notify on confirmations enabled
+        self.restart_node(1, ["-walletnotifyconfirmations=%s" % n_confirmations,
+                              "-walletnotify=echo %%s >> %s" % self.tx_filename])
+
+        # mine some more blocks (we should receive notifications on the ones above plus all but the last n_confirmations - 1 of these)
+        generated_after_restart = self.nodes[1].generate(block_count + n_confirmations - 1)[:-n_confirmations + 1]
+        wait_until(lambda: os.path.isfile(self.tx_filename) and os.stat(self.tx_filename).st_size >= (block_count * 65), timeout=10)
+
+        # get the list of block hashes that should now be confirmed
+        buried_blocks = generated_before_restart + generated_after_restart
```

Comment:
Unused symbol (causing the Travis failure). You can catch stuff like this locally by running ` ../../contrib/devtools/lint-python.sh` from within `test/functional`.

Reply:
Thanks! Updated